internal/engine.go

```diff
@@ -20,23 +20,45 @@ type Engine struct {
 	nolintMgr    *nolint.Manager
 	rules        []LintRule
 	defaultRules []LintRule
+	allRules     []LintRule
 }
 
 // NewEngine creates a new lint engine.
-func NewEngine(rootDir string, source []byte) (*Engine, error) {
+func NewEngine(rootDir string, source []byte, rules map[string]tt.ConfigRule) (*Engine, error) {
 	engine := &Engine{}
-	engine.initDefaultRules()
+	engine.applyRules(rules)
 
 	return engine, nil
 }
 
-// registerDefaultRules adds the default set of lint rules to the engine.
-func (e *Engine) registerDefaultRules() {
-	e.rules = append(e.rules, e.defaultRules...)
+func (e *Engine) applyRules(rules map[string]tt.ConfigRule) {
+	e.registerDefaultRules()
+	e.registerAllRules()
+
+	// Iterate over the rules and apply severity
+	for _, rule := range e.allRules {
+		if _, ok := rules[rule.Name()]; ok {
+			severity := rules[rule.Name()].Severity
+			appliedRule := e.findRule(rule.Name())
+			if appliedRule != nil {
+				// if severity is OFF, ignore the rule
+				if severity == tt.SeverityOff {
+					e.IgnoreRule(rule.Name())
+					continue
+				}
+				// set the severity of the rule
+				(*appliedRule).SetSeverity(severity)
+			} else {
+				// Add a new rule with the given severity
+				rule.SetSeverity(severity)
+				e.rules = append(e.rules, rule)
```

Comment:
this part dose not covered by test. could you add some case for this?

Reply:
added in https://github.com/gnoverse/tlin/pull/91/commits/27d198a6d85bc00e7bc295cf8e142195f42ec0df